examples/asr/structured/speech_to_text_structured.py

```diff
@@ -0,0 +1,165 @@
+
+# set global values
+cfg.model.repeat = 5
+cfg.model.separable = True
+
+LABELS = [
+    " ",
+    "a",
+    "b",
+    "c",
+    "d",
+    "e",
+    "f",
+    "g",
+    "h",
+    "i",
+    "j",
+    "k",
+    "l",
+    "m",
+    "n",
+    "o",
+    "p",
+    "q",
+    "r",
+    "s",
+    "t",
+    "u",
+    "v",
+    "w",
+    "x",
+    "y",
+    "z",
+    "'",
+]
+
+qn_15x5 = [
+    configs.JasperEncoderConfig(
+        filters=256,
+        repeat=1,
+        kernel=[33],
+        stride=[2],
+        separable=cfg.model.separable,
+        dilation=[1],
+        dropout=cfg.model.dropout,
+        residual=False,
+    ),
+    configs.JasperEncoderConfig(
+        filters=256,
+        repeat=1,
+        kernel=[33],
+        stride=[1],
+        separable=cfg.model.separable,
+        dilation=[1],
+        dropout=cfg.model.dropout,
+        residual=True,
+    ),
+    # ... repeat 14 more times
+    configs.JasperEncoderConfig(
+        filters=1024, repeat=1, kernel=[1], stride=[1], dilation=[1], dropout=cfg.model.dropout, residual=False,
+    ),
+]
+
+
+def main():
+    # Update values
+    # MODEL UPDATES
+    cfg.name = "Mini QuartzNet"
+    cfg.model.labels = LABELS
+
+    # train ds
+    cfg.model.train_ds.manifest_filepath = (
+        "/home/smajumdar/PycharmProjects/NeMo-som/examples/asr/an4/train_manifest.json"
+    )
+    cfg.model.train_ds.labels = LABELS
+    cfg.model.train_ds.sample_rate = cfg.model.sample_rate
+
+    # validation ds
+    cfg.model.validation_ds.manifest_filepath = (
+        "/home/smajumdar/PycharmProjects/NeMo-som/examples/asr/an4/test_manifest.json"
+    )
+    cfg.model.validation_ds.labels = LABELS
+    cfg.model.validation_ds.sample_rate = cfg.model.sample_rate
+
+    # del `test_ds` does not work!
+    # Refer - https://stackoverflow.com/questions/58119758/how-to-remove-dataclass-attributes
+    # Hydra/OmegaConf dont allow custom .asdict() methods either
+    # For now, explicitly set parameters
+    cfg.model.test_ds.sample_rate = cfg.model.sample_rate
+    cfg.model.test_ds.labels = cfg.model.labels
+
+    # preprocessor
+    cfg.model.preprocessor.sample_rate = cfg.model.sample_rate
+
+    # spec aug
+    cfg.model.spec_augment.rect_masks = 5
+    cfg.model.spec_augment.rect_freq = 50
+    cfg.model.spec_augment.rect_time = 120
+
+    # encoder
+    cfg.model.encoder.feat_in = cfg.model.preprocessor.features
+    cfg.model.encoder.activation = 'relu'
+    cfg.model.encoder.jasper = qn_15x5
+
+    # decoder
+    cfg.model.decoder.feat_in = qn_15x5[-1].filters
+    cfg.model.decoder.num_classes = len(LABELS)
+    cfg.model.decoder.vocabulary = LABELS
+
+    # optim
+    cfg.model.optim.name = 'novograd'
+    cfg.model.optim.lr = 0.01
+
+    cfg.model.optim.betas = [0.8, 0.5]
+    cfg.model.optim.weight_decay = 0.001
+
+    # sched
+    cfg.model.optim.sched.name = "CosineAnnealing"
+    cfg.model.optim.sched.warmup_steps = None
+    cfg.model.optim.sched.warmup_ratio = 0.01
```

Comment:
I forgot to mention it in the code, but theres an issue here. All of the "additional" configs dont actually get updated and provided to the ModelPT engine for optim and sched !

Ie if you run the code and look at the logs, youll see that `betas` is [0.95, 0.98] (the default for Novograd) and `weight_decay` is 0. Same for sched - `warmup_steps` and `warmup_ratio` are not provided as args.

```python
[NeMo I 2020-10-22 17:31:14 modelPT:587] Optimizer config = Novograd (
    Parameter Group 0
        amsgrad: False
        betas: (0.95, 0.98)
        eps: 1e-08
        grad_averaging: False
        lr: 0.01
        weight_decay: 0
    )
[NeMo I 2020-10-22 17:31:14 lr_scheduler:554] Scheduler "<nemo.core.optim.lr_scheduler.CosineAnnealing object at 0x7f7904106590>" 
    will be used during training (effective maximum steps = 148) - 
    Parameters : 
    (min_lr: 0.0
    last_epoch: -1
    max_steps: 1
```

Reply:
@ericharper @tkornuta-nvidia @okuchaiev @blisc This is another issue - the only way to override using yaml using `update_model_config`. Added a note about this